tasks/human_eval_006.rs

```diff
@@ -5,14 +5,120 @@ HumanEval/6
 /*
 ### VERUS BEGIN
 */
+use vstd::math::*;
 use vstd::prelude::*;
 
 verus! {
 
 // TODO: Put your solution (the specification, implementation, and proof) to the task here
+/// Given a space separated group of nested parentheses
+/// return a list of maximum depths of each group.
+pub open spec fn spec_nested_parens(s: Seq<char>) -> Option<Seq<int>> {
+    spec_nested_parens_helper(s, 0, 0, seq![])
+}
+
+/// Formalizes the problem as a rewrite system
+pub open spec fn spec_nested_parens_helper(
+    s: Seq<char>,
+    depth: int,
+    max_depth: int,
+    prev_depths: Seq<int>,
+) -> Option<Seq<int>>
+    decreases s.len(),
+{
+    if s.len() == 0 && depth == 0 {
+        if max_depth == 0 {
+            // (s, d, m, p) => Some(p) if s is empty and d is 0 and m == 0
+            Some(prev_depths)
+        } else {
+            // (s, d, m, p) => Some(p) if s is empty and d is 0 and m != 0
```

Comment:
This comment has the same `Some(p)` as above.  Should it be `Some(p + [max_depth])`?

Reply:
Fixed. Thanks for catching that!